WordPress/Classes/Services/ReaderCardService.swift

```diff
@@ -3,44 +3,69 @@ import Foundation
 class ReaderCardService {
     private let service: ReaderPostServiceRemote
     private let coreDataStack: CoreDataStack
+    private lazy var syncContext: NSManagedObjectContext = {
+        return coreDataStack.newDerivedContext()
+    }()
 
     init(service: ReaderPostServiceRemote = ReaderPostServiceRemote(wordPressComRestApi: WordPressComMockrestApi()),
          coreDataStack: CoreDataStack = ContextManager.shared) {
         self.service = service
         self.coreDataStack = coreDataStack
     }
 
-    func fetch(success: @escaping (Int, Bool) -> Void, failure: @escaping (Error?) -> Void) {
+    func fetch(page: Int = 1, success: @escaping (Int, Bool) -> Void, failure: @escaping (Error?) -> Void) {
         service.fetchCards(for: ["foo", "bar"],
-                           success: { [unowned self] cards in
-                            let syncContext = self.coreDataStack.newDerivedContext()
+                           success: { [weak self] cards in
 
-                            syncContext.perform {
-                                cards.forEach { remoteCard in
+                            guard let syncContext = self?.syncContext else {
+                                return
+                            }
+
+                            self?.syncContext.perform {
+
+                                if page == Constants.firstPage {
+                                    self?.clean()
+                                }
+
+                                cards.enumerated().forEach { index, remoteCard in
                                     guard remoteCard.type != .unknown else {
                                         return
                                     }
 
-                                    let card = ReaderCard(context: syncContext)
-
-                                    switch remoteCard.type {
-                                    case .post:
-                                        card.post = ReaderPost.createOrReplace(fromRemotePost: remoteCard.post, for: nil, context: syncContext)
-                                    default:
-                                        break
-                                    }
+                                    let card = ReaderCard(context: syncContext, from: remoteCard)
 
-                                    card.sortRank = Date().timeIntervalSince1970
+                                    // To keep the API order
+                                    card?.sortRank = Double((page * Constants.paginationMultiplier) + index)
                                 }
                             }
 
-                            self.coreDataStack.save(syncContext) {
+                            self?.coreDataStack.save(syncContext) {
                                 success(cards.count, true)
                             }
         }, failure: { error in
             failure(error)
         })
     }
+
+    private func clean() {
```

Comment:
Can we make this method name a bit more descriptive?

Reply:
Done in 01d9cd0dd3b5b686e23b1ceb52ad8cb662143fed